### PR TITLE
OHRM5X-1954: Develop claim - request action API

### PR DESCRIPTION
**Pull request by DulsaraNethmin**

### CODE REVIEW

File: installer/Migration/V5_4_0/permission/api.yaml [hunk 1]
```diff
@@ -12,5 +12,11 @@ apiv2_pim_my_info:
...
-    - { role: Admin, permission: { read: true, create: false, update: false, delete: false } }
-    - { role: ESS, permission: { read: true, create: false, update: false, delete: false } }
+    - {
+        role: Admin,
+        permission: { read: true, create: false, update: false, delete: false },
+      }
+    - {
+        role: ESS,
+        permission: { read: true, create: false, update: false, delete: false },
+      }
```
Comment (RajithaKumara) [inline installer/Migration/V5_4_0/permission/api.yaml:22]: not sure why this change

File: src/plugins/orangehrmClaimPlugin/Api/ClaimRequestActionAPI.php [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        WorkflowStateMachine::CLAIM_ACTION_APPROVE => 'APPROVE',
+        WorkflowStateMachine::CLAIM_ACTION_PAY => 'PAY',
+        WorkflowStateMachine::CLAIM_ACTION_CANCEL => 'CANCEL',
+        WorkflowStateMachine::CLAIM_ACTION_REJECT => 'REJECT'
+    ];
+
+    /**
+     * @OA\Put(
+     *     path="/api/v2/claim/requests/{requestId}/action",
+     *     tags={"Claim/Action"},
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/ClaimRequestActionAPI.php:66]:
```suggestion
     *     tags={"Claim/Requests"},
```
will put it under `Claim/Requests`

File: src/plugins/orangehrmClaimPlugin/Api/ClaimRequestActionAPI.php [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+                RequestParams::PARAM_TYPE_ATTRIBUTE,
+                self::PARAMETER_REQUEST_ID
+            );
+            $action = $this->getRequestParams()->getString(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_ACTION
+            );
+
+            $claimRequest = $this->getClaimRequest($requestId);
+            $actionIndex = array_search($action, self::ACTIONABLE_STATES_MAP);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/ClaimRequestActionAPI.php:105]:
```suggestion
            $actionIndex = array_flip(self::ACTIONABLE_STATES_MAP)[$action];
```
:thinking:

File: src/plugins/orangehrmClaimPlugin/test/Api/ClaimRequestActionAPITest.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+use OrangeHRM\Framework\Services;
+use OrangeHRM\Tests\Util\EndpointIntegrationTestCase;
+use OrangeHRM\Tests\Util\Integration\TestCaseParams;
+use OrangeHRM\Tests\Util\TestDataService;
+
+/**
+ * @group Claim
+ * @group APIv2
+ */
+class ClaimRequestActionAPITest extends EndpointIntegrationTestCase
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/test/Api/ClaimRequestActionAPITest.php:33]: cover not implemented methods also.
`delete`, `getOne`, `getValidationRuleForGetOne`, etc
```php
    public function testDelete(): void
    {
        $api = new ClaimRequestActionAPI($this->getRequest());
        $this->expectNotImplementedException();
        $api->delete();
    }

    public function testGetValidationRuleForDelete(): void
    {
        $api = new ClaimRequestActionAPI($this->getRequest());
        $this->expectNotImplementedException();
        $api->getValidationRuleForDelete();
    }
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: installer/Migration/V5_4_0/permission/api.yaml (hunk 1 above)
Comment (ManulMax) [reply installer/Migration/V5_4_0/permission/api.yaml]:
```suggestion
    - { role: Admin, permission: { read: false, create: false, update: true, delete: false } }
    - { role: ESS, permission: { read: false, create: false, update: true, delete: false } }
```

File: installer/Migration/V5_5_0/permission/api.yaml [hunk 5]
```diff
@@ -79,3 +79,22 @@ apiv2_claim_attachment:
...
+    delete: false
+  permissions:
+    - {
+      role: Admin,
+      permission: { read: false, create: false, update: true, delete: false },
+    }
+    - {
+      role: ESS,
+      permission: { read: false, create: false, update: true, delete: false },
+    }
```
Comment (ManulMax) [inline installer/Migration/V5_5_0/permission/api.yaml:100]:
```suggestion
    - { role: Admin, permission: { read: false, create: false, update: true, delete: false } }
    - { role: ESS, permission: { read: false, create: false, update: true, delete: false } }
```

**Review by ManulMax — CHANGES_REQUESTED** [review]

**Review by RajithaKumara — APPROVED** [review]
